askup_lti/models.py

```diff
@@ -0,0 +1,108 @@
+import hashlib
+import logging
+
+from django.conf import settings
+from django.contrib.auth import login
+from django.contrib.auth.models import User
+from django.db import models
+from django.db.models import fields
+import shortuuid
+
+log = logging.getLogger(__name__)
+
+
+def short_token():
+    """
+    Generate a hash that can be used as lti consumer key.
+    """
+    hash = hashlib.sha1(shortuuid.uuid().encode())
+    hash.update(settings.SECRET_KEY.encode())
+    return hash.hexdigest()[::2]
+
+
+class LtiProvider(models.Model):
+    """
+    Model to manage LTI consumers.
+
+    LMS connections.
+    Automatically generates key and secret for consumers.
+    """
+
+    consumer_name = models.CharField(max_length=255, unique=True)
+    consumer_key = models.CharField(max_length=32, unique=True, default=short_token)
+    consumer_secret = models.CharField(max_length=32, unique=True, default=short_token)
+
+    class Meta:
+        verbose_name = "LMS Platform"
+        verbose_name_plural = "LMS Platforms"
+
+    def __str__(self):
+        """
+        Return string representation of the LTIProvider.
+        """
+        return '<LtiProvider: {}>'.format(self.consumer_name)
+
+
+class LtiUser(models.Model):
+    """
+    Model to manage LTI users.
+    """
+
+    user_id = fields.CharField(max_length=255)
```

Comment:
Why not to use simple "id" here?
user_id is used a lot in other models of the project to link to the  Askup User.

Reply:
It is not a general id field it is identification which is received from the LTI request